tests/header_map.rs

```diff
@@ -86,6 +86,40 @@ fn drain() {
     }
 }
 
+#[test]
+fn drain_drop_immediately() {
+    // test mem::forgetting does not double-free
+
+    let mut headers = HeaderMap::new();
+    headers.insert("hello", "world".parse().unwrap());
+    headers.insert("zomg", "bar".parse().unwrap());
+    headers.append("hello", "world2".parse().unwrap());
+
+    let iter = headers.drain();
+    assert_eq!(iter.size_hint(), (2, Some(2)));
+    // not consuming `iter`
+}
+
+#[test]
+fn drain_forget() {
```

Comment:
Should there also be a test for `mem::forget`ting a `ValueDrain` returned by a `Drain`?